tests/test_0662-rntuple-stl-containers.py

```diff
@@ -0,0 +1,60 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/uproot4/blob/main/LICENSE
+
+import json
+import queue
+import sys
+
+import awkward as ak
+import numpy
+import pytest
+import skhep_testdata
+
+import uproot
+
+
+def test_rntuple_stl_containers():
+    filename = skhep_testdata.data_path("test_ntuple_stl_containers.root")
+    with uproot.open(filename) as f:
+        R = f["ntuple"]
+        assert R.keys() == [
+            "string",
+            "vector_int32",
+            "vector_vector_int32",
+            "vector_string",
+            "vector_vector_string",
+            "variant_int32_string",
+            "vector_variant_int64_string",
+            "tuple_int32_string",
+            "vector_tuple_int32_string",
+        ]
+        r = R.arrays()
+        assert ak.all(r["string"] == ["one", "two"])
+
+        assert r["vector_int32"][0] == [1]
+        assert ak.all(r["vector_int32"][1] == [1, 2])
+        assert r["vector_vector_int32"][0] == [[1]]
+
+        assert ak.all(r["vector_vector_int32"][1] == [[1], [2]])
+
+        assert r["vector_string"][0] == ["one"]
+        assert ak.all(r["vector_string"][1] == ["one", "two"])
+
+        assert ak.all(r["vector_vector_string"] == [["one"], [["one"], ["two"]]])
+        assert type(r["variant_int32_string"][0]) == numpy.int32
+        assert type(r["variant_int32_string"][1]) == str
+
+        assert r["vector_variant_int64_string"][0][0] == "one"
+        assert r["vector_variant_int64_string"][1][0] == "one"
+        assert r["vector_variant_int64_string"][1][1] == 2
+        assert type(r["vector_variant_int64_string"][1][1]) == numpy.int64
```

Comment:
For the variant tests, could you test the values, rather than the types? There's a 1% chance we'll change the type of

```python
r["variant_int32_string"][0]
```

from `np.int32` to `int` or `ak.Scalar` or something. (In Awkward v1, it was `int`, but I think we're doing the right thing now by making it `np.int32`. The `ak.Scalar` thing was talked about.)

It would be a more "what we care about" test to say

```python
        assert r["variant_int32_string"][0] == 1
        assert r["variant_int32_string"][1] == "two"
```

(though I don't know the actual values).